<?xml version="1.0" encoding="UTF-8"?>
<robot name="new_hand1">
  <joint name="right_index_abd" type="revolute">
    <origin xyz="-0.0260888 0.00334 0.0638849" rpy="1.5348927 -1.13552 3.0882027"/>
    <parent link="right_wrist_jointbody"/>
    <child link="right_index_abd_jointbody"/>
    <axis xyz="0. 0. 1."/>
    <limit lower="-1.04577" upper="0.24577" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_index_mcp" type="revolute">
    <origin xyz="0.02 0. -0. " rpy="3.1315856 1.5707927 -0.0100071"/>
    <parent link="right_index_abd_jointbody"/>
    <child link="right_index_mcp_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.6580599" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_index_pip" type="revolute">
    <origin xyz="0.0005 0. -0.038 " rpy="0.0000001 0. 0. "/>
    <parent link="right_index_mcp_jointbody"/>
    <child link="right_index_pip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.88496" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_middle_abd" type="revolute">
    <origin xyz="-0.0020001 0.00144 0.06872 " rpy="-1.6035732 -1.5707927 0.0327734"/>
    <parent link="right_wrist_jointbody"/>
    <child link="right_middle_abd_jointbody"/>
    <axis xyz="0. 0. 1."/>
    <limit lower="-0.64577" upper="0.64577" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_middle_mcp" type="revolute">
    <origin xyz="0.02 -0. 0. " rpy="-3.1361406 1.5707927 0.005452 "/>
    <parent link="right_middle_abd_jointbody"/>
    <child link="right_middle_mcp_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.58825" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_middle_pip" type="revolute">
    <origin xyz="0.0005 -0. -0.046 " rpy="-0.0000001 -0. -0.0000002"/>
    <parent link="right_middle_mcp_jointbody"/>
    <child link="right_middle_pip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.8674999" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_pinky_abd" type="revolute">
    <origin xyz="0.0405845 0.0064281 0.0459637" rpy="-1.6446526 -0.86149 0.2578627"/>
    <parent link="right_wrist_jointbody"/>
    <child link="right_pinky_abd_jointbody"/>
    <axis xyz="0. 0. 1."/>
    <limit lower="-0.12244" upper="1.1691" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_pinky_mcp" type="revolute">
    <origin xyz="0.02 -0. 0. " rpy="-3.1369045 1.5707927 0.004688 "/>
    <parent link="right_pinky_abd_jointbody"/>
    <child link="right_pinky_mcp_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.7104199" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_pinky_pip" type="revolute">
    <origin xyz="0.0005 0. -0.033 " rpy="0.0000001 0.0000001 -0.0000003"/>
    <parent link="right_pinky_mcp_jointbody"/>
    <child link="right_pinky_pip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.88496" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_ring_abd" type="revolute">
    <origin xyz="0.019521 0.0029782 0.0595448" rpy="-1.6056225 -1.30836 0.0871825"/>
    <parent link="right_wrist_jointbody"/>
    <child link="right_ring_abd_jointbody"/>
    <axis xyz="0. 0. 1."/>
    <limit lower="-0.47577" upper="0.8057699" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_ring_mcp" type="revolute">
    <origin xyz="0.02 -0. -0. " rpy="3.1384497 1.5707927 -0.0031431"/>
    <parent link="right_ring_abd_jointbody"/>
    <child link="right_ring_mcp_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.58825" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_ring_pip" type="revolute">
    <origin xyz="0.0005 0. -0.046 " rpy="0. 0. -0."/>
    <parent link="right_ring_mcp_jointbody"/>
    <child link="right_ring_pip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.34907" upper="1.8674999" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_thumb_abd" type="revolute">
    <origin xyz="-0. 0. -0." rpy="1.9198629 -0.0000002 1.5707964"/>
    <parent link="right_thumb_mcp_jointbody"/>
    <child link="right_thumb_abd_jointbody"/>
    <axis xyz="1. 0. 0."/>
    <limit lower="-1.08211" upper="0." effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_thumb_dip" type="revolute">
    <origin xyz="0.0005 -0. -0.033 " rpy="-0.0000002 -0.0000001 0.0000001"/>
    <parent link="right_thumb_pip_jointbody"/>
    <child link="right_thumb_dip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.8538399" upper="1.45" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_thumb_mcp" type="revolute">
    <origin xyz="-0.03021 0.01921 0.02551" rpy="2.6030226 0.05595 1.8711226"/>
    <parent link="right_wrist_jointbody"/>
    <child link="right_thumb_mcp_jointbody"/>
    <axis xyz="0. 0. 1."/>
    <limit lower="-0.87266" upper="0.87266" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_thumb_pip" type="revolute">
    <origin xyz="0. 0.0102606 0.0281908" rpy="-2.8120465 0.1172432 2.8120427"/>
    <parent link="right_thumb_abd_jointbody"/>
    <child link="right_thumb_pip_jointbody"/>
    <axis xyz="0. 1. 0."/>
    <limit lower="-0.7943999" upper="1.23" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="right_wrist" type="revolute">
    <origin xyz="0.00063 0. 0.10487" rpy="0. 0. 0."/>
    <parent link="hand_base"/>
    <child link="right_wrist_jointbody"/>
    <axis xyz="1. 0. 0."/>
    <limit lower="-0.6736" upper="0.8971999" effort="100." velocity="99.9999945"/>
  </joint>
  <joint name="root_joint" type="fixed">
    <origin xyz="0. 0. 0.0498567" rpy="0. 0. 0."/>
    <parent link="World"/>
    <child link="hand_base"/>
  </joint>
  <link name="World">
    <visual>
      <origin xyz="0. 0. 0." rpy="0. 0. 0."/>
      <geometry>
        <mesh filename="meshes/CollisionMesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="hand_base">
    <inertial>
      <origin xyz="0.0003396 -0.000909 0.0190988" rpy="0. 0. 0."/>
      <mass value="0.901"/>
      <inertia ixx="0.0015041" ixy="0.0000104" ixz="-0.0000052" iyy="0.0012534" iyz="-0.0000014" izz="0.0007962"/>
    </inertial>
    <visual>
      <origin xyz="0.0006951 -0.0001251 0.0231229" rpy="-0.0072522 -0.0043563 1.5339901"/>
      <geometry>
        <mesh filename="meshes/right_visual_tower_main_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006201 -0.0000037 0.0093393" rpy="1.5694791 0.0001519 -0.0006505"/>
      <geometry>
        <mesh filename="meshes/right_visual_tower_hull_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006175 0.0000044 0.0273773" rpy="-0.0000578 -1.5702411 1.5988764"/>
      <geometry>
        <mesh filename="meshes/right_visual_tower_text_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0010261 0.0008755 0.0375895" rpy="0.0663642 -0.105327 -0.7765911"/>
      <geometry>
        <mesh filename="meshes/right_collision_tower_main_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006188 0.0000071 0.0093062" rpy="1.5691929 -0.0000693 -0.0012917"/>
      <geometry>
        <mesh filename="meshes/right_collision_tower_hull_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006835 -0.0011251 0.0277161" rpy="-2.2396595 -1.5545672 -2.4480784"/>
      <geometry>
        <mesh filename="meshes/right_collision_tower_text_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_index_abd_jointbody">
    <inertial>
      <origin xyz="-0.007037 0. -0.0000333" rpy="0. 0. 0."/>
      <mass value="0.00598"/>
      <inertia ixx="0.0000002" ixy="0." ixz="-0." iyy="0.0000004" iyz="0." izz="0.0000004"/>
    </inertial>
    <visual>
      <origin xyz="0.0093015 -0.0000139 0.0002788" rpy="-0.0118206 1.506674 -3.141592 "/>
      <geometry>
        <mesh filename="meshes/right_visual_index_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0095218 -0.0000446 0.0001648" rpy="-2.21338 1.5147948 0.9289652"/>
      <geometry>
        <mesh filename="meshes/right_collision_index_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_index_mcp_jointbody">
    <inertial>
      <origin xyz="0.0012729 0.0000189 0.0154346" rpy="0. 0. 0."/>
      <mass value="0.0175"/>
      <inertia ixx="0.0000026" ixy="0." ixz="-0.0000001" iyy="0.0000026" iyz="0." izz="0.0000007"/>
    </inertial>
    <visual>
      <origin xyz="0.0012792 -0.0000432 -0.0160987" rpy="-3.1360765 -0.0351009 -0.0391045"/>
      <geometry>
        <mesh filename="meshes/right_visual_index_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0016617 -0.0000001 -0.0187639" rpy="0.0000002 0.8345405 0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_visual_index_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0013517 -0.0000112 -0.0156818" rpy="-3.1415925 -0.0428807 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_index_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0016774 0.0000055 -0.0187409" rpy="0.0000002 0.8259684 0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_collision_index_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_index_pip_jointbody">
    <inertial>
      <origin xyz="0.0006662 0.0000094 0.0181 " rpy="0. 0. 0."/>
      <mass value="0.01621"/>
      <inertia ixx="0.0000024" ixy="-0." ixz="0.0000003" iyy="0.0000024" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0010417 -0.0000784 -0.0117937" rpy="3.1071048 0.003439 -1.570915 "/>
      <geometry>
        <mesh filename="meshes/right_visual_index_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006178 0.0000003 -0.0229819" rpy="3.1415927 0.4563592 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_index_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0012645 0.0000164 -0.011472 " rpy="-3.139792 0. -1.5707964"/>
      <geometry>
        <mesh filename="meshes/right_collision_index_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006217 -0.0000024 -0.023021 " rpy="3.1415927 0.4457044 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_index_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_middle_abd_jointbody">
    <inertial>
      <origin xyz="-0.007037 0. -0.0000333" rpy="0. 0. 0."/>
      <mass value="0.00598"/>
      <inertia ixx="0.0000002" ixy="0." ixz="-0." iyy="0.0000004" iyz="0." izz="0.0000004"/>
    </inertial>
    <visual>
      <origin xyz="0.0093015 -0.0000139 0.0002788" rpy="-0.0118215 1.506674 3.1415923"/>
      <geometry>
        <mesh filename="meshes/right_visual_middle_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0095218 -0.0000446 0.0001648" rpy="-2.21338 1.5147948 0.9289652"/>
      <geometry>
        <mesh filename="meshes/right_collision_middle_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_middle_mcp_jointbody">
    <inertial>
      <origin xyz="0.0013716 0.0000191 0.0192504" rpy="0. 0. 0."/>
      <mass value="0.02106"/>
      <inertia ixx="0.0000042" ixy="0." ixz="-0.0000001" iyy="0.0000041" iyz="-0." izz="0.0000009"/>
    </inertial>
    <visual>
      <origin xyz="0.0012834 -0.000037 -0.0202272" rpy="-3.137028 -0.0276588 -0.037584 "/>
      <geometry>
        <mesh filename="meshes/right_visual_middle_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0017297 -0. -0.0229383" rpy="3.1415925 0.223125 -0.0000002"/>
      <geometry>
        <mesh filename="meshes/right_visual_middle_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0013394 0.0000068 -0.0201193" rpy="3.1409516 -0.0298785 -0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_collision_middle_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0017526 -0.0000027 -0.0228851" rpy="3.1415925 0.220061 -0.0000002"/>
      <geometry>
        <mesh filename="meshes/right_collision_middle_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_middle_pip_jointbody">
    <inertial>
      <origin xyz="0.0006662 0.0000094 0.0181 " rpy="0. 0. 0."/>
      <mass value="0.01621"/>
      <inertia ixx="0.0000024" ixy="-0." ixz="0.0000003" iyy="0.0000024" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0010417 -0.0000784 -0.0117938" rpy="3.1071048 0.003439 -1.570915 "/>
      <geometry>
        <mesh filename="meshes/right_visual_middle_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006178 0.0000003 -0.0229819" rpy="3.1415927 0.4563592 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_middle_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0012645 0.0000164 -0.011472 " rpy="-3.139792 0. -1.5707964"/>
      <geometry>
        <mesh filename="meshes/right_collision_middle_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006217 -0.0000024 -0.0230211" rpy="3.1415927 0.4457044 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_middle_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_pinky_abd_jointbody">
    <inertial>
      <origin xyz="-0.007037 0. -0.0000333" rpy="0. 0. 0."/>
      <mass value="0.00598"/>
      <inertia ixx="0.0000002" ixy="0." ixz="-0." iyy="0.0000004" iyz="0." izz="0.0000004"/>
    </inertial>
    <visual>
      <origin xyz="0.0093015 -0.0000139 0.0002788" rpy="-0.0118215 1.5066742 3.1415923"/>
      <geometry>
        <mesh filename="meshes/right_visual_pinky_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0095218 -0.0000446 0.0001648" rpy="-2.2133806 1.5147948 0.9289647"/>
      <geometry>
        <mesh filename="meshes/right_collision_pinky_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_pinky_mcp_jointbody">
    <inertial>
      <origin xyz="0.0012601 0.0000187 0.0129092" rpy="0. 0. 0."/>
      <mass value="0.01502"/>
      <inertia ixx="0.0000019" ixy="0." ixz="-0.0000001" iyy="0.0000018" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0012999 -0.0000072 -0.0136199" rpy="-3.1415925 -0.0433892 -0.0000003"/>
      <geometry>
        <mesh filename="meshes/right_visual_pinky_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0015424 -0.0000002 -0.0162596" rpy="0.0000003 1.2480961 -0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_visual_pinky_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.001356 -0.0001271 -0.0134463" rpy="-3.1345098 -0.047637 -0.0000003"/>
      <geometry>
        <mesh filename="meshes/right_collision_pinky_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0015528 -0.0000065 -0.0162467" rpy="0.0000003 1.2437718 -0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_collision_pinky_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_pinky_pip_jointbody">
    <inertial>
      <origin xyz="0.0006662 0.0000094 0.0181 " rpy="0. 0. 0."/>
      <mass value="0.01621"/>
      <inertia ixx="0.0000024" ixy="-0." ixz="0.0000003" iyy="0.0000024" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0010417 -0.0000784 -0.0117938" rpy="3.1071048 0.003439 -1.570915 "/>
      <geometry>
        <mesh filename="meshes/right_visual_pinky_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006178 0.0000003 -0.0229819" rpy="3.1415927 0.4563592 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_pinky_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0012645 0.0000164 -0.011472 " rpy="-3.139792 0. -1.5707964"/>
      <geometry>
        <mesh filename="meshes/right_collision_pinky_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006217 -0.0000024 -0.0230211" rpy="3.1415927 0.4457044 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_pinky_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_ring_abd_jointbody">
    <inertial>
      <origin xyz="-0.007037 0. -0.0000333" rpy="0. 0. 0."/>
      <mass value="0.00598"/>
      <inertia ixx="0.0000002" ixy="0." ixz="-0." iyy="0.0000004" iyz="0." izz="0.0000004"/>
    </inertial>
    <visual>
      <origin xyz="0.0093015 -0.0000139 0.0002788" rpy="-0.011821 1.506674 -3.1415925"/>
      <geometry>
        <mesh filename="meshes/right_visual_ring_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0095218 -0.0000446 0.0001648" rpy="-2.2133799 1.5147948 0.9289654"/>
      <geometry>
        <mesh filename="meshes/right_collision_ring_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_ring_mcp_jointbody">
    <inertial>
      <origin xyz="0.0013716 0.0000191 0.0192504" rpy="0. 0. 0."/>
      <mass value="0.02106"/>
      <inertia ixx="0.0000042" ixy="0." ixz="-0.0000001" iyy="0.0000041" iyz="-0." izz="0.0000009"/>
    </inertial>
    <visual>
      <origin xyz="0.0012834 -0.000037 -0.0202272" rpy="-3.137028 -0.0276588 -0.0375839"/>
      <geometry>
        <mesh filename="meshes/right_visual_ring_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0017297 -0. -0.0229383" rpy="-3.1415927 0.2231251 -0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_ring_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0013394 0.0000068 -0.0201193" rpy="3.1409516 -0.0298784 -0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_ring_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0017526 -0.0000027 -0.0228851" rpy="-3.1415927 0.220061 -0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_ring_pp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_ring_pip_jointbody">
    <inertial>
      <origin xyz="0.0006662 0.0000094 0.0181 " rpy="0. 0. 0."/>
      <mass value="0.01621"/>
      <inertia ixx="0.0000024" ixy="-0." ixz="0.0000003" iyy="0.0000024" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0010417 -0.0000784 -0.0117937" rpy="3.1071048 0.003439 -1.570915 "/>
      <geometry>
        <mesh filename="meshes/right_visual_ring_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006178 0.0000003 -0.0229819" rpy="3.1415927 0.4563592 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_ring_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0012645 0.0000164 -0.011472 " rpy="-3.139792 0. -1.5707964"/>
      <geometry>
        <mesh filename="meshes/right_collision_ring_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0006217 -0.0000024 -0.023021 " rpy="3.1415927 0.4457044 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_ring_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_thumb_abd_jointbody">
    <inertial>
      <origin xyz="-0.0099942 0.0000087 0.0000174" rpy="0. 0. 0."/>
      <mass value="0.00755"/>
      <inertia ixx="0.0000003" ixy="0." ixz="0." iyy="0.0000008" iyz="-0." izz="0.0000008"/>
    </inertial>
    <visual>
      <origin xyz="-0.0000952 0.0042424 0.0115234" rpy="2.8856726 -0.2342454 -0.7629122"/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="-0.0000952 0.0044057 0.0119384" rpy="2.8490174 -0.1803366 -0.5422454"/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_pp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_thumb_dip_jointbody">
    <inertial>
      <origin xyz="0.0021659 0.0000093 0.0105321" rpy="0. 0. 0."/>
      <mass value="0.0142"/>
      <inertia ixx="0.0000015" ixy="0." ixz="-0.0000001" iyy="0.0000015" iyz="0." izz="0.0000005"/>
    </inertial>
    <visual>
      <origin xyz="0.0017377 0.0000054 -0.0051273" rpy="3.1415927 -0.246983 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_dp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0027934 0.0000006 -0.016601 " rpy="0.0000001 0.4023979 0. "/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_dp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0020245 -0.0000368 -0.0055337" rpy="-2.8988683 0.0081907 -1.5687683"/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_dp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0027855 0.0000069 -0.0165575" rpy="0.0000001 0.3963953 0. "/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_dp_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_thumb_mcp_jointbody">
    <inertial>
      <origin xyz="0.0021099 0. 0.0034169" rpy="0. 0. 0."/>
      <mass value="0.00686"/>
      <inertia ixx="0.0000009" ixy="0." ixz="-0." iyy="0.0000007" iyz="0." izz="0.0000003"/>
    </inertial>
    <visual>
      <origin xyz="-0.0026673 -0.0000037 -0.0032022" rpy="3.1370592 0.0029216 -3.1415925"/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="-0.0025186 0.0000118 -0.0033326" rpy="3.1375663 0. -3.1415925"/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_mp_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_thumb_pip_jointbody">
    <inertial>
      <origin xyz="0.0012601 0.0000187 0.0129092" rpy="0. 0. 0."/>
      <mass value="0.01502"/>
      <inertia ixx="0.0000019" ixy="0." ixz="-0.0000001" iyy="0.0000018" iyz="0." izz="0.0000006"/>
    </inertial>
    <visual>
      <origin xyz="0.0012999 -0.0000072 -0.0136199" rpy="3.1415925 -0.0433893 0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0015424 -0.0000002 -0.0162596" rpy="-0.0000004 1.248096 -0.0000004"/>
      <geometry>
        <mesh filename="meshes/right_visual_thumb_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.001356 -0.0001271 -0.0134463" rpy="-3.13451 -0.0476372 0.0000001"/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_ip_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0015528 -0.0000065 -0.0162467" rpy="-0.0000004 1.2437717 -0.0000004"/>
      <geometry>
        <mesh filename="meshes/right_collision_thumb_ip_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
  <link name="right_wrist_jointbody">
    <inertial>
      <origin xyz="-0.0000674 0.0008855 0.0308785" rpy="0. 0. 0."/>
      <mass value="0.0267"/>
      <inertia ixx="0.0000078" ixy="0.0000002" ixz="0.0000005" iyy="0.0000171" iyz="-0.0000003" izz="0.0000115"/>
    </inertial>
    <visual>
      <origin xyz="-0.0004432 0.0017637 0.0268236" rpy="2.0117252 0.0471796 1.5313832"/>
      <geometry>
        <mesh filename="meshes/right_visual_palm_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0036733 0.0001697 0.0369756" rpy="1.7666059 0.1563942 1.6505203"/>
      <geometry>
        <mesh filename="meshes/right_visual_palm_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="-0.0004204 0.0015776 0.0267961" rpy="2.026338 0.0449174 1.5318477"/>
      <geometry>
        <mesh filename="meshes/right_collision_palm_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
    <visual>
      <origin xyz="0.0060138 0.0001463 0.0371503" rpy="1.8840619 0.1678501 1.6839923"/>
      <geometry>
        <mesh filename="meshes/right_collision_palm_skin_mesh_mesh.obj" scale="1. 1. 1."/>
      </geometry>
    </visual>
  </link>
</robot>
